### PR TITLE
Fixed Issue https://github.com/FreeCAD/FreeCAD-addons/issues/10

### PR DESCRIPTION
**Pull request by biosafetylvl5**
By downloading addons descriptions asynchronously, the issue was fixed. See InfoWorker. Moreover, a variety of syntactical changes were made in accordance with PEP8.

### CODE REVIEW

**Comment by biosafetylvl5** [comment]
In short, the descriptions are downloaded asynchronously rather than in response to a user clicking an addon

**Comment by yorikvanhavre** [comment]
Oops sorry I ended up merging partially your changes because I saw the issue report and not this PR. Also this commit changes a lot of stuff and it was unclear to me where the change is... I'll try to sort things out

File: addons_installer.FCMacro [hunk 1]
```diff
@@ -108,24 +107,40 @@ class AddonsInstaller(QtGui.QDialog):
...
+        else:
+            return self.repos
+
     def update(self):
-        self.update_worker = UpdateWorker()
+        self.info_worker = InfoWorker()
+        self.info_worker.addon_repos.connect(self.update_repos)
+        # self.info_worker.start()
+
+        self.update_worker = UpdateWorker(self.info_worker)
```
Comment (biosafetylvl5) [inline addons_installer.FCMacro]: Setup "InfoWorker" which fetches the repo information in the background, preventing crashes while improving use expirience

File: addons_installer.FCMacro [hunk 2]
```diff
@@ -170,65 +183,93 @@ class AddonsInstaller(QtGui.QDialog):
...
+
         self.progressbar_show.emit(False)
         self.stop = True
 
+        self.info_worker.repos = repos
+        self.info_worker.start()
 
-class ShowWorker(QtCore.QThread):
 
+class InfoWorker(QtCore.QThread):
```
Comment (biosafetylvl5) [inline addons_installer.FCMacro]: Fetches repo info in the background

**Comment by biosafetylvl5** [comment]
Sorry about the confusion, I added some comments to my commit that should hopefully clear things up. Most of the other changes are just slights adjustments for syntax, and aren't important for code performance.

**Comment by yorikvanhavre** [comment]
done with 0df6409

**Pull request closed without merging**